cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManagerFactory.java

```diff
@@ -29,14 +31,17 @@ public class RepositoryManagerFactory {
 
   private final SecureStore secureStore;
   private final CConfiguration cConf;
+  private final RepositoryStore repoStore;
 
   @Inject
-  RepositoryManagerFactory(SecureStore secureStore, CConfiguration cConf) {
+  RepositoryManagerFactory(SecureStore secureStore, CConfiguration cConf, RepositoryStore repoStore) {
     this.secureStore = secureStore;
     this.cConf = cConf;
+    this.repoStore = repoStore;
   }
 
-  public RepositoryManager create(NamespaceId namespace, RepositoryConfig repoConfig) {
-    return new RepositoryManager(secureStore, cConf, namespace, repoConfig);
+  public RepositoryManager create(NamespaceId namespace) throws RepositoryNotFoundException {
+    RepositoryMeta repositoryMeta = repoStore.getRepositoryMeta(namespace);
```

Comment:
I'm not sure that this is actually better than before, it makes it easy to accidentally introduce race conditions or inconsistencies in the future. 

For example, if we lookup the repository meta at the start of a long running operation, then call this later on, the repository meta might be different.

Reply:
@albertshau Could you pls elaborate a bit on the race conditions?

Compared to before, we pass in the `RepositoryConfig`, which could be error prone and not matching the `NamespaceId`. Now we look up by the `NamespaceId` and makes sure it's the correct config to use.

Reply:
I think now we don't create manager multiple times for the same rest call, do we? It would be something to look about when we start doing multipush

Reply:
@tivv yes we only create the manager once for one rest call.

Reply:
It's not just if we create multiple managers for the same operation, but also if we look up the RepositoryConfig earlier as well for whatever reason. Looking it up again here means the config can be different, or even if we're not worried about that, it's also additional load on the DB call and another call that can potentially fail.